Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -124,16 +128,17 @@ public async Task ValidateContestPassword(int id, bool official, string password
 
             if (participant == null)
             {
-                participant = await this.AddNewParticipantToContest(contest, model.IsOfficial, user.Id, user.IsAdmin);
+                participant = await this.AddNewParticipantToContestIfNotExists(contest, model.IsOfficial, user.Id, user.IsAdmin);
             }
 
-            if (model.IsOfficial &&
-                !await this.IsContestIpValidByContestAndIp(model.ContestId, model.UserHostAddress))
+            if (participant == null)
```

Comment:
Two identical if clauses in same method?

Reply:
I used the null coalescing operator in the first if clause and removed the second.